PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseStorageAPIsWithLengthParameter.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class UseStorageAPIsWithLengthParameter extends LocalInspectionTool {
+    @Nls
+    @NotNull
+    @Override
+    public String getDisplayName() {
+        return "Use Storage APIs With Length Parameter";
+    }
+
+    private static final List<String> METHODS_TO_CHECK = Arrays.asList("uploadFromFile", "upload", "uploadWithResponse");
```

Comment:
As we add more rules and as more APIs are added to SDKs, we'll need to restructure data like this so that it is easy to find and update. Think about how you'd want to structure these longer term for a future PR update - either as a config file or having all of these static final variables grouped together in a common part of the class/source so it's easier for future contributors to maintain.

Reply:
We should consider having a configuration class to hold the APIs to be flagged.

Reply:
made rulesConfig.json. could only work with absolute file paths